### DISCUSSION TITLE
How do I test changes to a formula?

### DISCUSSION THREAD

**Discussion by SSteve** [Homebrew/discussions#2640] [Writing Formulae/Casks] [answered]
I'm getting an error ("ArgumentError: comparison of String with 8 failed") building llvm on MacOS Mojave. I know Mojave isn't supported, so I'm trying to fix it myself. Installing with debug and verbose flags helped me track it down to [this line](https://github.com/Homebrew/homebrew-core/blob/c662296a962cd9fcf3d7bd9280cefbdeaf63f7d5/Formula/llvm.rb#L322). I think it should be `compat_version = Integer(xcode_version) < 8 ? "1" : "2"`. I'm pretty sure changes I make to llvm.rb before installing are overwritten. How can I run the install with my version of the formula?

Building llvm takes hours so there's a long time between starting the install and getting the error message. Is there any way to keep the build products and re-run the install with them?

**Comment by carlocab** [comment, marked as answer]
Oops, sorry for that. This should be fixed by Homebrew/homebrew-core#91436.

As for your question: you should be able to just do `brew edit llvm`, make your changes, and then do `brew install llvm`.

**Comment by carlocab** [comment]
This should be fixed now. Do `brew update` to pull in the fix.

**Reply by SSteve** [reply]
It worked great, thanks!
